pagopa/pn-frontend
# Feat/pn 7345

## Short description
create logic and component to able cancel notification
## List of changes proposed in this pull request
- decomment part of notification detail with logic of cancel notificatio
- create new method and type for adapt cancel status of notification in timeline
- create new modal for understand cancel notification when payment is enabled 
- create test for all 

## How to test
- Login with grossini - palermo -> search for NRJP-NZRW-LDTL-202308-L-1 -> go to detail and check that notification has cancellation in progress
- Login with grossini - palermo -> search for HYTD-ERPH-WDUE-202308-H-1 -> go to detail and check that notification is cancelled
-  Login with grossini - palermo -> search for PELM-VYNK-XVGV-202308-R-1 -> go to detail and click on cancel button -> check that cancellation goes well
- Login with grossini - palermo -> search for XJTU-DJLJ-MEGE-202308-N-1 -> go to detail and click on cancel button -> check that cancellation returns 409
- Login with grossini - palermo -> go to detail (iun different from above) and click on cancel button -> check that cancellation returns 500
